net: mdns: adding MDNS unicast response confirm rfc6732

SynchronicIT opened the pull request:
Conform rfc6732 a mDNS responder should answer clients which are not using the mDNS port in the source address with unicast  UDP to the same port.

Fixes: #81657

SynchronicIT committed:
net: mdns: adding MDNS unicast response confirm rfc6732

Conform rfc6762  a mDNS responder should answer clients
which are not using the mDNS port in the source address
with unicast  UDP to the same port as described in chapter 6.7

Fixes: #81657

Signed-off-by: Vincent van der Locht <vincent@synchronicit.nl>

rlubos reviewed (commented):
Looks reasonable and aligned with RFC.
However, the commit message should rather reference [rfc67**6**2](https://datatracker.ietf.org/doc/html/rfc6762), not rfc6732? Also it would be nice to mention the RFC chapter this functionality is described (ch. 6.7 I believe?), for future reference.

SynchronicIT committed:
tests: net: mdns: updated interface to match unicast MDNS function

To add the functionality conform rfc6762 chapter 6.7  the interface for
setup_dst_addr(...) has changed. This patch fixes the interface mismatch

Signed-off-by: Vincent van der Locht <vincent@synchronicit.nl>

SynchronicIT commented on subsys/net/lib/dns/mdns_responder.c:
```diff
@@ -194,25 +196,36 @@ static int set_ttl_hop_limit(int sock, int level, int option, int new_limit)
 }
 
 int setup_dst_addr(int sock, sa_family_t family,
```
I've added the fix for the test. Would it be needed to extend the test too, to validate the unicast use-case? For IPv4, I know how to do so, for IPv6 I'm not familiar with the address range structure to pick a correct local test address.
